docs/bigquery/snippets.py

```diff
@@ -2579,6 +2579,178 @@ def test_query_no_cache(client):
+    ]
+    external_config.schema = [
+        bigquery.SchemaField('name', 'STRING'),
+        bigquery.SchemaField('post_abbr', 'STRING')
+    ]
+    external_config.options.skip_leading_rows = 1  # optionally skip header row
+    table_id = 'us_states'
+    job_config = bigquery.QueryJobConfig()
+    job_config.table_definitions = {table_id: external_config}
+
+    # Example query to find states starting with 'W'
+    sql = 'SELECT * FROM {} WHERE name like "W%"'.format(table_id)
+
+    query_job = client.query(sql, job_config=job_config)  # API request
+
+    w_states = list(query_job)  # Waits for query to finish
+    print('There are {} states with names starting with W.'.format(
+        len(w_states)))
+    # [END bigquery_query_external_gcs_temp]
+    assert len(w_states) == 4
+
+
+def test_query_external_gcs_permanent_table(client, to_delete):
+    dataset_id = 'query_external_gcs_{}'.format(_millis())
+    dataset = bigquery.Dataset(client.dataset(dataset_id))
+    client.create_dataset(dataset)
+    to_delete.append(dataset)
+
+    # [START bigquery_query_external_gcs_perm]
+    # from google.cloud import bigquery
+    # client = bigquery.Client()
+    # dataset_id = 'my_dataset'
+
+    # Configure the external data source
+    dataset_ref = client.dataset(dataset_id)
+    table_id = 'us_states'
+    schema = [
+        bigquery.SchemaField('name', 'STRING'),
+        bigquery.SchemaField('post_abbr', 'STRING')
+    ]
+    table = bigquery.Table(dataset_ref.table(table_id), schema=schema)
+    external_config = bigquery.ExternalConfig('CSV')
+    external_config.source_uris = [
+        'gs://cloud-samples-data/bigquery/us-states/us-states.csv',
+    ]
+    external_config.options.skip_leading_rows = 1  # optionally skip header row
+    table.external_data_configuration = external_config
+
+    # Create a permanent table linked to the GCS file
+    table = client.create_table(table)  # API request
+
+    # Example query to find states starting with 'W'
+    sql = 'SELECT * FROM {}.{} WHERE name like "W%"'.format(
+        dataset_id, table_id)
+
+    query_job = client.query(sql)  # API request
+
+    w_states = list(query_job)  # Waits for query to finish
+    print('There are {} states with names starting with W.'.format(
+        len(w_states)))
+    # [END bigquery_query_external_gcs_perm]
+    assert len(w_states) == 4
+
+
+def test_query_external_sheets_temporary_table(client):
+    # [START bigquery_query_external_sheets_temp]
+    # [START bigquery_auth_drive_scope]
+    # from google.cloud import bigquery
+
+    import google.auth
+
+    # Create credentials with Drive & BigQuery API scopes
+    # Both APIs must be enabled for your project before running this code
+    credentials, project = google.auth.default(scopes=[
+        'https://www.googleapis.com/auth/drive',
+        'https://www.googleapis.com/auth/bigquery',
+    ])
+    client = bigquery.Client(credentials=credentials)
```

Comment:
Should probably pass `project` in, too. Otherwise `Client` is going to have to call `google.auth.default` again internally to get the same project out.